ui/src/lib/gql/project/hook.ts

```diff
@@ -0,0 +1,88 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+
+import { useGraphQLContext } from "@flow/lib/gql";
+import { CreateProject, DeleteProject, GetProjects, Project } from "@flow/types";
+
+import { CreateProjectInput, DeleteProjectInput } from "../__gen__/graphql";
+
+export enum ProjectQueryKeys {
+  GetProjects = "getProjects",
+}
+
+export const useProject = () => {
+  const graphQLContext = useGraphQLContext();
+  const queryClient = useQueryClient();
+
+  const createProjectMutation = useMutation({
+    mutationFn: async (input: CreateProjectInput) => {
+      const data = await graphQLContext?.CreateProject({ input });
+      return { project: data?.createProject?.project };
+    },
+    onSuccess: project =>
+      // TODO: Maybe update cache and not refetch?
+      queryClient.invalidateQueries({
+        queryKey: [ProjectQueryKeys.GetProjects, project.project?.workspaceId],
+      }),
+  });
+
+  const useGetProjectsQuery = (workspaceId: string) =>
+    useQuery({
+      queryKey: [ProjectQueryKeys.GetProjects, workspaceId],
+      queryFn: async () => {
+        // TODO: Should have pagination
+        const data = await graphQLContext?.GetProjects({ workspaceId, first: 20 });
+        if (!data) return {};
+        const {
+          projects: { nodes, ...rest },
+        } = data;
+        return { projects: nodes as Project[], meta: rest };
```

Comment:
Don't typecast here either. projects will still have all of the variables that come from the query, but will just be invisible to everyone, which isn't ideal.
Instead, map through nodes and create new objects that match `Project`.
That way everything is consistent and we can be more deliberate what we are sending to the app 🙏🏻

Reply:
This is Done!